pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -269,7 +269,12 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
                 for (int i = messagesForC; i < entriesWithSameKeyCount; i++) {
                     Entry entry = entriesWithSameKey.get(i);
                     long stickyKeyHash = getStickyKeyHash(entry);
-                    addMessageToReplay(entry.getLedgerId(), entry.getEntryId(), stickyKeyHash);
+                    // add to replay only if message is not received by previous consumer
+                    boolean isDeleted = cursor
+                            .isMessageDeleted(new PositionImpl(entry.getLedgerId(), entry.getEntryId()));
+                    if (!isDeleted && !isEntryPendingAck(entry.getLedgerId(), entry.getEntryId())) {
+                        addMessageToReplay(entry.getLedgerId(), entry.getEntryId(), stickyKeyHash);
```

Comment:
If you add part of messages to the replay queue, do you need to reset the read position of the cursor? Otherwise, how the skipped entries can be read again?